### PR TITLE
Feature 10744 add bar graphs to sample dashboard

### LINKED ISSUE #10744: Add  bar graphs to sample dashboard [3]
### Feature Description

A graph should be added to the sample dashboard to provide an overview of samples documented and tested over time. The structure should be similar to the epidemiological curve on the surveillance dashboard. 

### Proposed Change

- [x] add heading `Sample Bar Chart`
- [x] add a legend for each colour for enums  for `final laboratory test`, group by: Positive, Negative, Pending, indeterminate, Not done
- [x] add a graph which shows the number of samples (y-axis) based on the selected options in the "grouping" button; there needs to be a dropdown that allows the user to choose between report date, sample collection date (i.e. `sampeldatetime`) and most relevant date (default); most relevant date uses the earliest pathogen test date among all pathogen tests associated with the sample, and falls back to the sample collection date if there are no tests or there is no test with a pathogen test date; then the user can group by day, epi week or month as shown in the graphic below (x-axis) 
- [x] add button to enlarge/scale down the graph to the top right
- [x] add button Chart Context Menu to download the graph (formats: PNG; JPEG; PDF; SVG; CSV; XLS) on the top right below the button to enlarge the graph
- [x] add button to change grouping between day, epi-week or month and to decide if at least seven entries should be shown

**OPEN QUESTION:**
~~Shouldn't we also consider relevance date here? According to https://github.com/hzi-braunschweig/SORMAS-Project/issues/10721, we allow filtering for relevance date on the samples dashboard, only considering report date in the graph seems inconsistent.~~

image

image

### Acceptance Criteria

The weekly sample overview is implemented as described above. 

### Additional Information

The data used in the global filter  is the report data of the parent entity of the sample:
* cases:  case report date
* Contact: report date
*  event participant:  report date of event
~~mmunization participant: report data of immunization or that of the associated case (reduced immunization)~~
* travel return:  return data of travel return

### PR DESCRIPTION
**Pull request by leventegal-she**
Fixes #10744

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11691

**Review by carina29 — CHANGES_REQUESTED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11691

File: sormas-ui/src/main/java/de/symeda/sormas/ui/dashboard/diagram/AbstractEpiCurveBuilder.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+				+ "color: (Highcharts.theme && Highcharts.theme.dataLabelsColor) || 'white' } } },");
+
+		hcjs.append(buildEpiCurve(datesGroupedBy, dashboardDataProvider)).append(", ");
+
+		//@formatter:off
+		hcjs.append("exporting: {\n" +
+				"        buttons: {\n" +
+				"            contextButton: {\n" +
+				"                menuItems: [\n" +
+				"                    'printChart',\n" +
```
[the remark this answers is not in the document]
Comment (leventegal-she) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/dashboard/diagram/AbstractEpiCurveBuilder.java:91]: the ticket does not request printing the whole dashboard page. It's about downloading the chart in different formats using the menu icon on the top right corner of the chart
